### ISSUE TITLE
to_csv() design and implementation

### ISSUE THREAD

**Issue by CommonClimate** [LinkedEarth/Pyleoclim_util#291]
**Is your feature request related to a problem? Please describe.**
Exporting pyleo.Series to CSV is one thing PaleoCube funded us to do.

**Describe the solution you'd like**

with pandas, exporting the data will be trivial. The question is : what to do about the metadata. Options:

- ignore them (ouch)
- include as header, which can be skipped when reading 
- CSV on the Web [recommendation](https://www.w3.org/TR/tabular-data-model/#embedded-metadata) (unclear how it helps here)
- put metadata in a [JSON string](https://stackoverflow.com/a/48706350) embedded in last column header
- emulate [Xarray](https://xarray-extras.readthedocs.io/en/latest/api/csv.html), though it's unclear what is done with the metadata in that case.

**Comment by khider** [comment]
Clarifying metadata here. We're talking about the parameters of the object, correct? 

It's not difficult to export two table within the csv, one that contains the parameter info and the other the data itself. For instance, for a Series object, the first table would contain info such as time_name, time_unit, value_name, value_unit... and the the second table the values for time and value. Exporting is not difficult. 

Reimporting from a csv is not difficult per se as long as the parameter names do not vary from one version to the other. Or we will need to maintain compatibility under the hood, which might require a lot of maintenance effort as is the case with JSON.

**Comment by CommonClimate** [comment]
> Clarifying metadata here. We're talking about the parameters of the object, correct?

"Parameters" isn't disambiguating anything much ... I am referring to everything that is not `value` or `time`, as in: 
```
@property
    def metadata(self):
        return dict(
            time_unit = self.time_unit,
            time_name = self.time_name,
            value_unit = self.value_unit,
            value_name = self.value_name,
            label = self.label,
            lat = self.lat,
            lon = self.lon,
            archiveType = self.archiveType,
            dataset_name = self.dataset_name,
            log = self.log
        )

```

> It's not difficult to export two table within the csv, one that contains the parameter info and the other the data itself. For instance, for a Series object, the first table would contain info such as time_name, time_unit, value_name, value_unit... and the the second table the values for time and value. Exporting is not difficult.

I like the idea of exporting to two different sheets, but I don't see that option in the [pandas method](https://pandas.pydata.org/docs/reference/api/pandas.Series.to_csv.html), so we'll need to hack it. 

 > Reimporting from a csv is not difficult per se as long as the parameter names do not vary from one version to the other. 
 
These are fixed dictionary keys, and they will remain so so long as we do not change the nomenclature of Series. Which brings me to this point: we should decide whether we use archiveType or archive_type, and dataSetName or dataset_name (or should it be dataset_id?).

**Comment by CommonClimate** [comment]
Apparently, CSV files can only have one sheet: https://stackoverflow.com/a/58236552
What do you want to do? 
1.  stick with CSV, lossy export
1. Export to xlsx with 2 sheets
1. export to CSV + something else for metadata (e.g. JSON, which will look suspiciously like lipd...).  

For option 1,  we can stuff as much metadata into the file name and column headers, like:  name the file  `label + '.csv'` , and the column  headers `time_name + '(' + time_unit  + ')' and  `value_name + '(' + value_unit  + ')' . We will lose things like lat/lon and archiveType, but at least those 5 pieces of metadata won't be wasted. Thoughts?

**Comment by khider** [comment]
That's why I said two tables. And we can have some hint that separates the two on the same sheet. We could do something like NOAA with `#` to indicate metadata. And then the table.

**Comment by CommonClimate** [comment]
Prototype available in [5ed03fa](https://github.com/LinkedEarth/Pyleoclim_util/commit/5ed03fa7767f256489f837eafef4fc1edbcd4dd2), with this result:



### | Series metadata
written by | Pyleoclim 1.0.0ba
time_unit | kyr BP
time_name | Age
value_unit | ppm
value_name | $CO_2$
label | EPICA Dome C CO2
lat |  
lon |  
archiveType |  
importedFrom
log | ({0: 'clean_ts', 'applied': True, 'verbose': False},)
### | end metadata
Age [kyr BP] | $CO_2$ [ppm]
-0.05103 | 368.02
-0.048 | 361.78
-0.04628 | 359.65
-0.04441 | 357.11
-0.04308 | 353.95
-0.04231 | 353.72
-0.04113 | 352.42
-0.03997 | 350.81
-0.03941 | 349.8
-0.03897 | 349.28
-0.03807 | 347.6
-0.03721 | 345.44
-0.03662 | 344.72
-0.03614 | 343.97

**Comment by CommonClimate** [comment]
The round-trip to/from CSV is now working as of e1d64c68b99505dd8f56f6cbe87337325e90d58a. The last remaining glitch is that going Series --> CSV --> Series yields very small differences (order 10^{-14}) in the time axis, so the two Series objects (before vs after) are not identical. Marco suggests we should not worry about perfectly roundtripping, but the `Series.equals` method should probably not error on such small differences.   To be resolved in issue #316 .

**Comment by CommonClimate** [comment]
This still needs a test; waiting on advice from Kim on the best way to do that with load_datasets().

**Comment by CommonClimate** [comment]
Resolved by #324
